src/main/java/io/asyncer/r2dbc/mysql/message/client/LocalInfileResponse.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright 2024 asyncer.io projects
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.asyncer.r2dbc.mysql.message.client;
+
+import io.asyncer.r2dbc.mysql.ConnectionContext;
+import io.asyncer.r2dbc.mysql.internal.util.NettyBufferUtils;
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.ByteBufAllocator;
+import io.r2dbc.spi.R2dbcNonTransientResourceException;
+import io.r2dbc.spi.R2dbcPermissionDeniedException;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+import reactor.core.publisher.SynchronousSink;
+
+import java.nio.file.InvalidPathException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.concurrent.atomic.AtomicReference;
+
+import static io.asyncer.r2dbc.mysql.internal.util.AssertUtils.requireNonNull;
+
+/**
+ * A message considers as a chunk of a local in-file data.
+ */
+public final class LocalInfileResponse implements SubsequenceClientMessage {
+
+    private final int envelopeId;
+
+    private final String path;
+
+    private final SynchronousSink<?> errorSink;
+
+    public LocalInfileResponse(int envelopeId, String path, SynchronousSink<?> errorSink) {
+        requireNonNull(path, "path must not be null");
+
+        this.envelopeId = envelopeId;
+        this.path = path;
+        this.errorSink = errorSink;
+    }
+
+    @Override
+    public Flux<ByteBuf> encode(ByteBufAllocator allocator, ConnectionContext context) {
+        return Flux.defer(() -> {
+            int bufferSize = context.getLocalInfileBufferSize();
+            AtomicReference<Throwable> error = new AtomicReference<>();
+
+            return Mono.<Path>create(sink -> {
+                try {
+                    Path safePath = context.getLocalInfilePath();
+                    Path file = Paths.get(this.path);
+
+                    if (safePath == null || file.startsWith(safePath)) {
```

Comment:
`safePath` is `null` means `allowLoadLocalInfileInPath = null`, 

```suggestion
                    if (safePath != null && file.startsWith(safePath)) {
```